### ISSUE TITLE
FreeIPA docker run error

### ISSUE THREAD

**Issue by guillermocol** [freeipa/freeipa-container#391]
Hello, 

I have been building the image for Dockerfile.centos-7 and it went smooth.  All the steps of the Dockerfile were successful. 
The issue arises when we try to run the docker image. The error that I am getting is the following one:
                                 No handlers could be found for logger "ipapython.admintool"

The steps taken are the following ones:

1. docker build -t freeipa-server -f Dockerfile.centos-7 .
2. setsebool -P container_manage_cgroup 1
3. mkdir /var/lib/ipa-data
4. docker run -ti -h $HOSTNAME --read-only         -v /sys/fs/cgroup:/sys/fs/cgroup:ro         -v /var/lib/ipa-data:/data:Z         freeipa-server ipa-server-install

Out of curiosity I have been running also the following command: `tests/run-partial-tests.sh Dockerfile.centos-7`. I add the last lines of the execution:

```
Mar 31 15:22:41 ipa.example.test systemd[1]: Starting GSSAPI Proxy Daemon...
Mar 31 15:22:41 ipa.example.test systemd[1]: Starting Login Service...
Mar 31 15:22:41 ipa.example.test systemd[1]: systemd-journald.service: Couldn't add fd to fd store: Operation not permitted
Mar 31 15:22:41 ipa.example.test systemd[1]: systemd-journald.service: Couldn't add fd to fd store: Operation not permitted
Mar 31 15:22:41 ipa.example.test systemd[1]: systemd-journald.service: Couldn't add fd to fd store: Operation not permitted
Mar 31 15:22:41 ipa.example.test systemd[1]: systemd-journald.service: Couldn't add fd to fd store: Operation not permitted
Mar 31 15:22:41 ipa.example.test systemd[1]: Started RPC bind service.
Mar 31 15:22:41 ipa.example.test systemd[1]: network.service: control process exited, code=exited status=6
Mar 31 15:22:41 ipa.example.test systemd[1]: Failed to start LSB: Bring up/down networking.
Mar 31 15:22:41 ipa.example.test systemd[1]: Unit network.service entered failed state.
Mar 31 15:22:41 ipa.example.test systemd[1]: network.service failed.
Mar 31 15:22:41 ipa.example.test systemd[1]: Starting Cleanup of Temporary Directories...
Mar 31 15:22:41 ipa.example.test systemd[1]: Reached target Network.
Mar 31 15:22:41 ipa.example.test systemd[1]: Reached target Network is Online.
Mar 31 15:22:41 ipa.example.test systemd[1]: Starting Notify NFS peers of a restart...
Mar 31 15:22:41 ipa.example.test systemd[1]: systemd-journald.service: Couldn't add fd to fd store: Operation not permitted
Mar 31 15:22:41 ipa.example.test sm-notify[54]: Version 1.3.0 starting
Mar 31 15:22:41 ipa.example.test systemd[1]: Started Notify NFS peers of a restart.
Mar 31 15:22:41 ipa.example.test systemd[1]: Started Cleanup of Temporary Directories.
Mar 31 15:22:41 ipa.example.test systemd[1]: Started Login Service.
Mar 31 15:22:41 ipa.example.test systemd-logind[52]: New seat seat0.
Mar 31 15:22:41 ipa.example.test systemd[1]: Started GSSAPI Proxy Daemon.
Mar 31 15:22:41 ipa.example.test systemd[1]: Reached target NFS client services.
Mar 31 15:22:41 ipa.example.test systemd[1]: Reached target Remote File Systems (Pre).
Mar 31 15:22:41 ipa.example.test systemd[1]: Reached target Remote File Systems.
Mar 31 15:22:41 ipa.example.test systemd[1]: Starting Permit User Sessions...
Mar 31 15:22:41 ipa.example.test systemd[1]: Started Permit User Sessions.
Mar 31 15:22:41 ipa.example.test systemd[1]: Reached target Login Prompts.
Mar 31 15:22:41 ipa.example.test systemd[1]: Reached target Multi-User System.
Mar 31 15:22:41 ipa.example.test systemd[1]: Starting Update UTMP about System Runlevel Changes...
Mar 31 15:22:41 ipa.example.test systemd[1]: Started Update UTMP about System Runlevel Changes.
Mar 31 15:22:41 ipa.example.test systemd[1]: Startup finished in 731ms.
sys-fs-fuse-connections.mount loaded failed failed FUSE Control File System
var-lib-nfs-rpc_pipefs.mount  loaded failed failed RPC Pipe File System
network.service               loaded failed failed LSB: Bring up/down networking
● network.service - LSB: Bring up/down networking
   Loaded: loaded (/etc/rc.d/init.d/network; bad; vendor preset: disabled)
   Active: failed (Result: exit-code) since Wed 2021-03-31 15:22:41 UTC; 2s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 46 ExecStart=/etc/rc.d/init.d/network start (code=exited, status=6)

Mar 31 15:22:41 ipa.example.test systemd[1]: Starting LSB: Bring up/down networking...
Mar 31 15:22:41 ipa.example.test systemd[1]: network.service: control process exited, code=exited status=6
Mar 31 15:22:41 ipa.example.test systemd[1]: Failed to start LSB: Bring up/down networking.
Mar 31 15:22:41 ipa.example.test systemd[1]: Unit network.service entered failed state.
Mar 31 15:22:41 ipa.example.test systemd[1]: network.service failed.
● sys-fs-fuse-connections.mount - FUSE Control File System
   Loaded: loaded (/usr/lib/systemd/system/sys-fs-fuse-connections.mount; static; vendor preset: disabled)
   Active: failed (Result: exit-code) since Wed 2021-03-31 15:22:41 UTC; 3s ago
    Where: /sys/fs/fuse/connections
     What: fusectl
     Docs: https://www.kernel.org/doc/Documentation/filesystems/fuse.txt
           http://www.freedesktop.org/wiki/Software/systemd/APIFileSystems
  Process: 24 ExecMount=/bin/mount fusectl /sys/fs/fuse/connections -t fusectl (code=exited, status=32)

Mar 31 15:22:41 ipa.example.test mount[24]: mount: permission denied
Mar 31 15:22:41 ipa.example.test systemd[1]: sys-fs-fuse-connections.mount mount process exited, code=exited status=32
Mar 31 15:22:41 ipa.example.test systemd[1]: Failed to mount FUSE Control File System.
Mar 31 15:22:41 ipa.example.test systemd[1]: Unit sys-fs-fuse-connections.mount entered failed state.
Warning: Journal has been rotated since unit was started. Log output is incomplete or unavailable.
● var-lib-nfs-rpc_pipefs.mount - RPC Pipe File System
   Loaded: loaded (/usr/lib/systemd/system/var-lib-nfs-rpc_pipefs.mount; static; vendor preset: disabled)
   Active: failed (Result: exit-code) since Wed 2021-03-31 15:22:41 UTC; 3s ago
    Where: /var/lib/nfs/rpc_pipefs
     What: sunrpc
  Process: 43 ExecMount=/bin/mount sunrpc /var/lib/nfs/rpc_pipefs -t rpc_pipefs (code=exited, status=32)

Mar 31 15:22:41 ipa.example.test systemd[1]: Mounting RPC Pipe File System...
Mar 31 15:22:41 ipa.example.test systemd[1]: var-lib-nfs-rpc_pipefs.mount mount process exited, code=exited status=32
Mar 31 15:22:41 ipa.example.test mount[43]: mount: permission denied
Mar 31 15:22:41 ipa.example.test systemd[1]: Failed to mount RPC Pipe File System.
Mar 31 15:22:41 ipa.example.test systemd[1]: Unit var-lib-nfs-rpc_pipefs.mount entered failed state.
1a2
> sys-fs-fuse-connections.mount
```

In case it helps this additional information.

I have been trying to do different kind of docker run, but I always get the same error message:
       No handlers could be found for logger "ipapython.admintool"

My current Operative System where those commands were run from is Linux Centos 7

Any comments would be appreciated.
Thanks for your help,

**Comment by adelton** [comment]
Please try to rerun the test command as
```
TRAVIS=yes tests/run-partial-tests.sh Dockerfile.centos-7
```
It dies early because it sees that `sys-fs-fuse-connections.mount` fail, pretending we are on Travis CI should get you further along.

**Comment by guillermocol** [comment]
Hi Adelton, seems to get better results with this test command:

 ---> e414b9ffc034
Removing intermediate container 508b75a2aca5
Successfully built e414b9ffc034
Sending build context to Docker daemon 370.7 kB
Step 1/2 : FROM local/freeipa-server-test:centos-7
 ---> e414b9ffc034
Step 2/2 : VOLUME /var/log/journal
 ---> Running in dfc37f9ac46a
 ---> 2bb358fd97b5
Removing intermediate container dfc37f9ac46a
Successfully built 2bb358fd97b5
+ docker run --name freeipa-server-container-centos-7 -d -h ipa.example.test --tmpfs /run --tmpfs /tmp -v /sys/fs/cgroup:/sys/fs/cgroup:ro --sysctl net.ipv6.conf.all.disable_ipv6=0 local/freeipa-server-test-addons:centos-7
d58ac406549a560db7313c4072ecca73b3ebeac5f702b92cd0533c3ea5636a1a
Executing tests/systemd-container-diff.sh freeipa-server-container-centos-7 list-dependencies-centos-7.out docker-diff-minimal-centos-7.exceptions docker-diff-minimal-centos-7.out
● ipa.example.test
    State: running
     Jobs: 0 queued
   Failed: 0 units
    Since: Thu 2021-04-01 08:01:49 UTC; 353ms ago
   CGroup: /system.slice/docker-d58ac406549a560db7313c4072ecca73b3ebeac5f702b92cd0533c3ea5636a1a.scope
           ├─ 1 /usr/sbin/init
           ├─30 systemctl status --no-pager -l
           └─system.slice
             ├─dbus.service
             │ └─22 /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation
             └─systemd-journald.service
               └─20 /usr/lib/systemd/systemd-journald
running
default.target
● ├─dbus.service
● │ ├─dbus.socket
● │ │ └─sysinit.target
● │ └─basic.target
● │   └─sysinit.target
● ├─systemd-journald.service
● │ ├─systemd-journald.socket
● │ └─systemd-journald.socket
● └─systemd-tmpfiles-setup.service
ls: cannot access /var/log/journal/d58ac406549a560db7313c4072ecca73/system.journal: No such file or directory
-rw-r-----. 1 root root 8388608 Apr  1 08:01 /run/log/journal/d58ac406549a560db7313c4072ecca73/system.journal
ls: cannot access /var/log/journal/d58ac406549a560db7313c4072ecca73/system.journal: No such file or directory
C /etc
C /etc/machine-id
C /var
C /var/log
C /var/log/btmp
C /var/log/wtmp
1a2,3
> C /var


But during the installation, what can I do with this: "No handlers could be found for logger "ipapython.admintool""

Thank you

**Comment by adelton** [comment]
You might want to prepend a couple more `#` characters to the `# test: systemd-container-diff.sh tests/systemd-container-diff.sh ...` line to skip this failing test.

As for the `No handlers could be found for logger "ipapython.admintool"` -- I've never seen such error and googling around it does not seem to be a known generic problem. The whole purpose of the `tests/run-partial-tests.sh` debugging is to make sure that the installation as it's been tested during development and in our CI works on your setup, before you start to play with the interactive `ipa-server-install`s.

**Comment by guillermocol** [comment]
Hi Adelton,

I get the following error now running the tests after commenting those lines:

Step 2/2 : VOLUME /var/log/journal
 ---> Running in c4cf53b43b9b
 ---> caff2e3e9184
Removing intermediate container c4cf53b43b9b
Successfully built caff2e3e9184
+ docker run --name freeipa-server-container-centos-7 -d -h ipa.example.test --tmpfs /run --tmpfs /tmp -v /sys/fs/cgroup:/sys/fs/cgroup:ro --sysctl net.ipv6.conf.all.disable_ipv6=0 local/freeipa-server-test-addons:centos-7
e22ad06e2c796d5c3515ce45482c675256f3734392e2d415b34d71d2afc4ce62
/usr/bin/docker-current: Error response from daemon: oci runtime error: container_linux.go:235: starting container process caused "process_linux.go:339: running prestart hook 2 caused \"error running hook: exit status 1, stdout: , stderr: \"".

About the ipapython.admintool not sure how I will be able to sort it. No much information about it : (

Thank you for your help,

**Comment by guillermocol** [comment]
Do the tests create a docker service that can be started? I see I can run there ipa-server-install

**Comment by adelton** [comment]
It might be necessary to force-remove the oci-systemd-hook package from the host to overcome that `oci runtime error`.

**Comment by adelton** [comment]
I'm not sure what you mean by a docker service. The test create containers with testing environments.

**Comment by guillermocol** [comment]
Hi Adelton, the docker service with freeipa worked out in the test container, but not in the one I try to build.

**In run-partial-tests.sh**:
TEST_SCRIPT=$( sed -n '$s/^# test:\s*//;T;s/\( \|$\)/ 'freeipa-server-container-$SUFFIX' /;p' "$DOCKERFILE.part" )
	if [ -n "$TEST_SCRIPT" ] ; then
		$docker build -t "local/freeipa-server-test:$SUFFIX" -f "$DOCKERFILE.part" .
		echo "FROM local/freeipa-server-test:$SUFFIX" > "$DOCKERFILE.part.addons"
		sed -n 's/# test-addon:\s*//;T;p' "$DOCKERFILE.part" >> "$DOCKERFILE.part.addons"
		$docker build -t "local/freeipa-server-test-addons:$SUFFIX" -f "$DOCKERFILE.part.addons" .
		$docker rm -f freeipa-server-container-$SUFFIX > /dev/null 2>&1 || :

**This created a test container: TRAVIS=yes tests/run-partial-tests.sh Dockerfile.centos-7** 
This freeipa-server-test-addons was up and I was able to run the ipa-server-install however, I don't know why the Dockerfile.centos-7 build fails for me.

**docker build -t freeipa-server -f Dockerfile.centos-7 .**    
  
(No handlers could be found for logger "ipapython.admintool")

**Comment by adelton** [comment]
The `tests/run-partial-tests.sh` in its latest stage should run the `ipa-server-install` automatically, there is no need to run it manually, and in fact, I'd expect it to fail and say that FreeIPA is already configured.

You said before that `docker build` passed and `docker run` failed. Now you say `docker build` fails for you. Which way is it? What is the context of that `No handlers could be found for logger "ipapython.admintool"` message? We've already established that you see that error, I have no idea why because I was not able to reproduce it yet, and repeating the same single error line while adding confusion when you see it does not really help.

**Comment by guillermocol** [comment]
Sorry for the misunderstanding Adelton I wanted to say run the docker service, the build worked out.

docker build Successful
docker run Not successful ( No handlers could be found for logger "ipapython.admintool")

The test was failing, but the docker service was running with no issue after running the run-partial-tests.sh
I am trying to understand how in that docker service (testing docker service) the ipa-server-install worked out with no issue.

**Comment by adelton** [comment]
So you say the `No handlers could be found for logger "ipapython.admintool"` message is the only output you get when you run that `docker run` command?

**Comment by adelton** [comment]
There is no python3 in CentOS 7.

If you want install stuff to the container, you should not run at as `--read-only`.

**Comment by adelton** [comment]
Do you get the same error when you use the image from Docker Hub?

What does
```
# docker run --rm -ti freeipa/freeipa-server:centos-7 bash -c 'python /usr/lib/python2.7/site-packages/ipapython/admintool.py'
# echo $?
```
print?

What does
```
# rpm -q docker container-selinux oci-systemd-hook
```
print?

**Comment by guillermocol** [comment]
Hi Adelton, thank you for the guidance, I will tested and let you know details :)

**Comment by guillermocol** [comment]
I got the following error running the docker run -rm ...

Digest: sha256:ac53890ee386f10d18468957d93bb5197e9e0045125ef89d8c1ac6c21fab122e
Status: Downloaded newer image for docker.io/freeipa/freeipa-server:centos-7
/usr/bin/docker-current: Error response from daemon: oci runtime error: container_linux.go:235: starting container process caused "process_linux.go:339: running prestart hook 2 caused \"error running hook: exit status 1, stdout: , stderr: \"".

echo $? 125


rpm -q docker container-selinux oci-systemd-hook
docker-1.13.1-203.git0be3e21.el7.centos.x86_64
container-selinux-2.119.2-1.911c772.el7_8.noarch
oci-systemd-hook-0.2.0-1.git05e6923.el7_6.x86_64

I was able to run it using this:

docker run --env oci-systemd-hook=disabled --rm -ti -h freeipa.test.com  \
        -v /sys/fs/cgroup:/sys/fs/cgroup:ro \
        -v /var/lib/ipa-data:/data:Z \
        freeipa-server-gcm ipa-server-install

but I think then will not modify systemd services in the docker container. It didn't give error the installation, but checking if it goes through.
